.NET/Microsoft.Recognizers.Text.DateTime/Parsers/BaseSetParser.cs

```diff
@@ -221,12 +221,20 @@ private DateTimeResolutionResult ParseEach(IDateTimeExtractor extractor, IDateTi
             match = this.config.SetWeekDayRegex.Match(text);
             if (match.Success)
             {
-                var trimmedText = text.Remove(match.Index, match.Length);
-                trimmedText = trimmedText.Insert(match.Index, match.Groups["weekday"].ToString());
-                ers = extractor.Extract(trimmedText, refDate);
-                if (ers.Count == 1 && ers.First().Length == trimmedText.Length)
+                if (match.Value.EndsWith("s"))
```

Comment:
Instead of this if and using inheritance, it might be better to have all configs use composition and internally have a class to handle set extract and set parse. This one one base class can be used by default (with the contents of this if) and Italian can instantiate its own.
Then this piece of this method is just a call to config.SetWeekDayParser(). Similarly for the extractor.
You can remove the Draft mark after that.

Reply:
Perhaps call the class with both methods something like SetHandler.